nft_ingester/src/program_transformers/bubblegum/db.rs

```diff
@@ -333,8 +347,10 @@ where
         )
         .build(DbBackend::Postgres);
 
+    // Do not overwrite changes that happened after decompression (asset.seq = 0).
+    // Do not overwrite changes from a later Bubblegum instruction.
     query.sql = format!(
-        "{} WHERE (NOT asset.was_decompressed) AND (excluded.seq >= asset.seq OR asset.seq IS NULL)",
+        "{} WHERE (asset.seq != 0 AND excluded.seq >= asset.seq) OR asset.seq IS NULL",
```

Comment:
So the `was_decompressed` is actually just like a one-time-use sequence number for use when indexing compression instructions.  Because the actual Bubblegum`decompress_v1` instruction has no tree event / no sequence number, this was the solution I previously came up with to mitigate ordering issues with `decompress_v1`.

For example if there's a `mint_v1`, and a `decompress_v1` indexed out of order, the `was_decompressed` flag is meant to prevent `mint_v1` from overwriting the info written by `upsert_asset_with_compression_info` during `decompress_v1` indexing.

However, in main today, the `was_decompressed` usage in both `upsert_asset_with_seq` and `upsert_asset_with_leaf_info` is subject to the same kind of race condition you pointed out with the flawed Version 2 approach for `update_metadata`.  This concept is fixed by 1) not setting the overall `seq` to 0 in decompress, and 2) bundling the  `upsert_asset_with_leaf_info_for_decompression` and `upsert_asset_with_compression_info` in a single transaction.

This is all separate from the checks on `asset.seq != 0`, which is actually set to 0 in Token Metadata `save_v1_asset` and is the replacement for the `slot_updated` checks that were previously used for this purpose.  I didn't like the slot_updated checks because I think its possible to have multiple bubblegum transactions in the same slot, and we would want to index all of them.

Reply:
Sync'd offline and realized that there will always be a Token Metadata `save_v1_asset` as part of a Bubblegum `decompress_v1` instruction.  Therefore, there is no intermediate state where something can be decompressed in Bubblegum, but not yet indexed by account-based indexing for Token Metadata.  Because of that we don't need a separate `was_decompressed` flag, which was essentially a single-use sequence number for indexing compression info with Bubblegum transactions.  We can instead just use the overall `asset.seq`.